wsmaster/codenvy-hosted-system-ram/src/main/java/com/codenvy/service/systemRam/DockerBasedSystemRamLimitChecker.java

```diff
@@ -0,0 +1,141 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.service.systemRam;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.Inject;
+
+import org.eclipse.che.commons.lang.Size;
+import org.eclipse.che.commons.schedule.ScheduleDelay;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.SystemInfo;
+import org.everrest.core.impl.provider.json.JsonUtils;
+import org.everrest.websockets.WSConnectionContext;
+import org.everrest.websockets.message.ChannelBroadcastMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import javax.websocket.EncodeException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+import static com.codenvy.service.systemRam.SystemRamLimitStatus.SYSTEM_RAM_LIMIT_EXCEEDED;
+import static com.codenvy.service.systemRam.SystemRamLimitStatus.SYSTEM_RAM_LIMIT_NOT_EXCEEDED;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Implementation of {@link SystemRamLimitChecker}
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+public class DockerBasedSystemRamLimitChecker implements SystemRamLimitChecker{
```

Comment:
@vinokurig consider the following:

Separate this class on 2 different components:
- first(inteface based) for retrieving information about system ram, which can be implemented for different systems
- second(certain without an interface) for scheduling and notifying the clients about system status when necessary, it will inject and reuse the first component to perform checks